src/App.tsx

```diff
@@ -9,63 +9,55 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+
 import { Counter } from './features/counter/Counter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as postsActions } from './features/postsSlice';
+import { actions as selectedPostActions } from './features/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const dispatch = useAppDispatch();
+  const { author } = useAppSelector((state) => state.author);
+  const { items, loaded, hasError } = useAppSelector((state) => state.posts);
+  const { selectedPost } = useAppSelector((state) => state.selectedPost);
 
   function loadUserPosts(userId: number) {
-    setLoaded(false);
+    dispatch(postsActions.setLoaded(false));
 
     getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
+      .then((postsFromServer) => {
+        dispatch(postsActions.setItems(postsFromServer));
+      })
+      .catch(() => dispatch(postsActions.setHasError(true)))
+      .finally(() => dispatch(postsActions.setLoaded(true)));
```

Comment:
It's not the way you handle async logic in Redux. Read more about `createAsyncThunk` or RTK Query. It was mentioned in videos and in theory. Change it everywhere.